internal/abci/abci.go

```diff
@@ -573,21 +573,28 @@ func (a *AbciApp) FinalizeBlock(ctx context.Context, req *abciTypes.RequestFinal
 	defer done(ctx)
 
 	// wait group to wait at the end of the function for all logs to be received
-	var wg sync.WaitGroup
-	wg.Add(1)
+	logsDone := make(chan error, 1)
 	go func() {
+		defer close(logsDone)
+
 		// we enforce that the cumulative size of logs is less than 1KB
 		// per tx. This is a work-around until we have gas costs to protect
 		// against log spam.
 		for {
 			log, ok := <-logs
-			if !ok {
-				break
+			if !ok { // empty and closed (normal completion)
+				return // logsDone receiver gets nil error
+			}
+			if log == "" {
+				// The DB has shut down with active subscribers. Fail.
+				logsDone <- errors.New("premature notice stream termination")
+				return
 			}
 			txid, notice, err := parse.ParseNotice(log)
 			if err != nil {
-				// should be deterministic so nbd to not halt here
-				a.log.Errorf("failed to parse notice: %v", err)
+				// will still be deterministic so nbd to not halt here
+				a.log.Errorf("failed to parse notice (%.20s...): %v", log, err)
+				continue // since txid is invalid and won't match any result.TxHash
```

Comment:
Pls double check me on this `continue` @brennanjl .  I tend to agree it not necessary to halt, but I don't think we should attempt to use the `txid` as a map key below, just continue to receive in the next iteration of the loop

Reply:
Yeah this seems correct.